include/diy/mpi/io.hpp

```diff
@@ -70,12 +67,62 @@ namespace io
 }
 }
 
+diy::mpi::io::file::
+file(const communicator& comm, const std::string& filename, int mode)
+: comm_(comm)
+{
+#ifndef DIY_NO_MPI
+  MPI_File_open(comm, const_cast<char*>(filename.c_str()), mode, MPI_INFO_NULL, &fh);
+#else
+  (void) filename; (void) mode;
```

Comment:
Why no `DIY_UNSUPPORTED_MPI_CALL` here?

I'm not sure I agree that we can't support `diy::mpi::io::file` in the `DIY_NO_MPI` setting. Most of the operations (unlike the subarrays in BOV) are fairly straightforward to implement. I guess it's Ok for now, but we should reconsider this in the future.

My first comment stands: why not throw right from the constructor?

Reply:
good catch..just oversight..I added `DIY_UNSUPPORTED_MPI_CALL` after a few steps in when I was updating the code.